presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TableLayoutRewriter.java

```diff
@@ -56,10 +59,11 @@ public TableLayoutRewriter(Metadata metadata, Session session, SymbolAllocator s
 
     public PlanNode planTableScan(TableScanNode node, Expression predicate)
     {
+        Expression deterministicPredicate = filterDeterministicConjuncts(predicate);
```

Comment:
> Add correctness logic from AddExchanges to TableLayoutRewriter

to we have any test cases exposing the previously incorrect logic?

Reply:
I think we should remove this commit. It does not add much (nothing?): 
1. without this change non deterministic predicates should go to `decomposedPredicate.getRemainingExpression(),`  (line 97), so we do not need to filter deterministic predicates here.
2. layouts should always have requested columns, see the comment below

Reply:
Agreed